proxy/proxy_test.go

```diff
@@ -148,11 +152,29 @@ func runHTTPTest(t *testing.T, tc *testCase) {
 	// Authorization header set.
 	client := &http.Client{}
 	url := fmt.Sprintf("http://%s/http/test", testProxyAddr)
-	resp, err := client.Get(url)
+
+	req, err := http.NewRequest(method, url, nil)
+	require.NoError(t, err)
+
+	if method == "POST" {
+		req.Header.Add("Content-Type", "application/json")
+		req.Body = io.NopCloser(strings.NewReader(`{}`))
+	}
+
+	resp, err := client.Do(req)
 	require.NoError(t, err)
 
 	require.Equal(t, "402 Payment Required", resp.Status)
 
+	bodyContent, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Equal(t, "payment required\n", string(bodyContent))
+
+	require.Equal(
+		t, fmt.Sprintf("%d", len(bodyContent)),
+		resp.Header.Get("Content-Length"),
```

Comment:
Field `resp.ContentLength` can be used, to avoid using `fmt.Sprintf("%d",`

Go documentation says:

>	// When
>	// Header values are duplicated by other fields in this struct (e.g.,
>	// ContentLength, TransferEncoding, Trailer), the field values are
>	// authoritative.

https://pkg.go.dev/net/http#Response

There are two other instances below in the code.

Reply:
Makes sense, thanks! Fixed.